### ISSUE TITLE
Guidepup announcement is different from my local VoiceOver 

### ISSUE THREAD

**Issue by iamskok** [guidepup/guidepup#40]
Thanks a lot for coming up with such an incredible tool! It's truly a game-changer. 

It's a miss on my side or something else, but the announcement I get from Guidepup differs from what I get manually executing the same actions in the same browser. I'm trying to test this [dialog component](https://i1gvw3.csb.app/). When I run the test manually, I get dialog title and description announcements associated via `aria-labelledby` and `aria-describedby`. However, when I run Guidepup, I get back only the dialog title.

Video recording of the manual test

https://user-images.githubusercontent.com/5145359/213947112-2bcfdefb-2b60-435a-8dea-1bce57251150.mp4

Guidepup Test

```ts
import { expect } from '@playwright/test'
import test from './voiceover-test'
import type { VoiceOver } from '@guidepup/guidepup/lib/macOS/VoiceOver/VoiceOver'

function delay (ms: number) {
  return new Promise((resolve) => setTimeout(resolve, ms))
}

async function waitForWebContentAnnouncement (voiceOver) {
  for (let i = 0; i < 10; i++) {
    const itemText = await voiceOver.itemText()

    if (itemText?.includes('web content')) {
      return
    }

    await delay(50)
  }

  throw new Error('web content could not be found')
}

const moveToNextAndSpeak = async (voiceOver: VoiceOver, caseSensitive = false): Promise<string> => {
  await voiceOver.perform(voiceOver.keyboard.commands.moveToNext)
  const lastSpokenPhrase = await voiceOver.lastSpokenPhrase()
  return caseSensitive ? lastSpokenPhrase : lastSpokenPhrase.toLowerCase()
}
const moveToNextUntil = async (voiceOver: VoiceOver, str): Promise<void> => {
  while (await moveToNextAndSpeak(voiceOver) !== str) {
    continue
  }
}

test.describe('Dialog VoiceOver', () => {
  test('I can navigate to dialog demo', async ({
    page,
    voiceOver,
  }) => {
    await page.goto('https://i1gvw3.csb.app', {
      waitUntil: 'domcontentloaded',
    })

    const buttonSelector = '[data-a11y-dialog-show="my-dialog"]'
    const [buttonElement] = await page.locator(buttonSelector).all()
    await expect(buttonElement).toBeVisible()
    await waitForWebContentAnnouncement(voiceOver)
    await voiceOver.interact()

    await moveToNextUntil(voiceOver, 'open the dialog window button')
    await voiceOver.perform(voiceOver.keyboard.commands.performDefaultActionForItem)
    expect(await voiceOver.lastSpokenPhrase()).toContain('Subscribe to our newsletter web dialog with 2 items')
  })
})
```

Output from `voiceOver.spokenPhraseLog()`:

```
[
  'In a11y-dialog (v7) demo web content main',
  'heading level 1 a11y-dialog',
  'link Documentation ↗',
  '—',
  'link GitHub ↗',
  '—',
  'Open the dialog window button',
  'Subscribe to our newsletter web dialog with 2 items'
]
```

**Comment by cmorten** [comment]
Hey @iamskok 👋 

I think this is a valid issue, to confirm the entry / text that you're expecting but missing is:

![VoiceOver viewer pannel with text: Fill in your email address to receive our newsletter!, You are currently on a web dialogue, inside web content. To interact with items in this group, press Control-Option-...](https://user-images.githubusercontent.com/11313985/214013226-d32bf07a-71f7-417f-af27-86f4bc796058.png)

???

I believe the issue is with the log capture for VoiceOver which is only grabbing the first announcement after an action and not waiting for subsequent content - coincidentally something I've addressed for the WIP NVDA support.

Pertinent code https://github.com/guidepup/guidepup/blob/main/src/LogStore.ts#L37.

The API for getting VoiceOver spoken content is a pull mechanism, so likely will need to update this logic to poll for the last spoken content until it stabilises, i.e. hasn't changed over X checks / T timeout.

A related / tangent piece would be to look into VoiceOver Utility settings for Verbosity and Speech (suspect other things as well) - certainly the speech rate impacts the poll time, so could foresee a world where it's driven from the settings. Some thought needed in general because any local changes to these settings won't be reflected in CI by the tool, perhaps we need an API for that (but now getting off-topic, one for future).

**Comment by iamskok** [comment]
Thanks for your prompt response!

> ![](https://user-images.githubusercontent.com/11313985/214013226-d32bf07a-71f7-417f-af27-86f4bc796058.png)

Correct, this is the missing part of the text.

> I believe the issue is with the log capture for VoiceOver which is only grabbing the first announcement after an action and not waiting for subsequent content

I also have a feeling that this is what's happening.

> certainly the speech rate impacts the poll time

I believe the speech rate is the only thing I ever changed, and it was set to 30. Just to rule out this option, I increased the rate to 90 and the `voiceOver.spokenPhraseLog()` hasn't changed.

> Some thought needed in general because any local changes to these settings won't be reflected in CI by the tool, perhaps we need an API for that (but now getting off-topic, one for future).

An API for settings sounds like a good idea!

**Comment by iamskok** [comment]
@cmorten, thanks for looking into this issue so quickly! However, after pulling the new version, my test started failing with `Error: web content could not be found`. I also tried a few things out but nothing seems to work. The test was taking too long to execute and there were no actual announcements. I prepared a reproduction if you will have a chance to check it out? https://github.com/iamskok/guidepup-repro
